File: myscripts/python/testscripts/ThreadTest.java
import java.io.*;
import java.net.*;

public class ThreadTest
{
    public static void main(String argv[]) throws Exception
    {
        int portno;
        int buffer_size = 1000;
        int total = 0;

        for (int i = 0; i < 500; ++i) {
            MyThread t1 = new MyThread(i, 1000, 500000);
            t1.start();
        }

        if (argv.length > 0) {
            try {
                portno = Integer.parseInt(argv[0]);
                Socket clientSocket = new Socket("localhost", portno);
                byte[] buf = new byte[buffer_size];
                DataOutputStream outToServer = new DataOutputStream(clientSocket.getOutputStream());
                outToServer.write(buf);
                System.out.println("Sent " + buffer_size + " bytes\n");
                total += buffer_size;
                clientSocket.close ();
                System.out.println("Sent " + total + " bytes\n");
            } catch (NumberFormatException e) {
                System.err.println("Argument " + argv[1] + " must be an integer.");
                System.exit(1);
            }
        }
    }
}

File: myscripts/python/testscripts/threadtest.py
import threading
import socket
import sys

total = 0
buffer_size = 1000

# called by each thread
def print_id(s, n, m):
    for ss in xrange(s,m,n):
        print "[ID " + str(s) + "] " + str(ss)

for i in xrange(1000):
    t = threading.Thread(target=print_id, args = (i, 500, 500000))
    t.daemon = True
    t.start()

if len (sys.argv) > 1:
    portno = int(sys.argv[1])
    # Create a TCP/IP socket
    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

    # Connect the socket to the port where the server is listening
    server_address = ("localhost", portno)
    print >>sys.stderr, 'connecting to %s port %s' % server_address
    sock.connect(server_address)

    try:
        # Send data
        message = bytearray([1 for b in range(buffer_size) ])
        print >>sys.stderr, 'sending %d bytes' % len(message)
        sock.sendall(message)

        total += len(message)
        print >>sys.stderr, 'sent %d total bytes' % total

    finally:
        print >>sys.stderr, 'closing socket'
        sock.close()
